File: user-app/lib/data/api/auth_api.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../core/network/api_client.dart';
import '../models/auth_response.dart';

final authApiProvider = Provider<AuthApi>((ref) {
  return AuthApi(ref.read(apiClientProvider));
});

class AuthApi {
  final ApiClient _client;

  AuthApi(this._client);

  Future<AuthResponse> login(String email, String password) async {
    final response = await _client.dio.post(
      '/auth/login',
      data: {'email': email, 'password': password},
    );
    return AuthResponse.fromJson(response.data as Map<String, dynamic>);
  }

  Future<AuthResponse> register({
    required String email,
    required String password,
    required String fullName,
    String? phone,
  }) async {
    final response = await _client.dio.post(
      '/auth/register',
      data: {
        'email': email,
        'password': password,
        'fullName': fullName,
        if (phone != null && phone.isNotEmpty) 'phone': phone,
        'roles': ['USER'],
      },
    );
    return AuthResponse.fromJson(response.data as Map<String, dynamic>);
  }

  Future<void> logout() async {
    try {
      await _client.dio.post('/auth/logout');
    } catch (_) {
      // Best effort – always clear local state
    }
  }

  Future<Map<String, dynamic>> getMe() async {
    final response = await _client.dio.get('/auth/me');
    return response.data as Map<String, dynamic>;
  }
}

File: hospital-dashboard/src/api/auth.api.ts
import apiClient from './client.ts';
import type { LoginRequest, LoginResponse, User } from '../types/index.ts';

export const login = (data: LoginRequest) =>
  apiClient.post<LoginResponse>('/api/auth/login', data);

export const logout = () =>
  apiClient.post('/api/auth/logout');

export const refreshToken = (token: string) =>
  apiClient.post<LoginResponse>('/api/auth/refresh', { refreshToken: token });

export const getMe = () =>
  apiClient.get<User>('/api/auth/me');
